js/main.js

```diff
@@ -32,6 +32,81 @@ function initNonVueButtons () {
   document.getElementById('button_configuration').addEventListener('click', function (evt) {
     Store.togglePopup('configuration')
   })
+  $(function () {
+    $('input[name="daterange"]').daterangepicker({
+      opens: 'left',
+      'timePicker': true,
+      'timePicker24Hour': true,
+      'showCustomRangeLabel': false,
+      'locale': {
+        'format': 'DD/MM/YYYY HH:mm',
+        'firstDay': 1
+      },
+      ranges: {
+        'Last 5 minutes': [,],
+        'Last 15 minutes': [,],
+        'Last 30 minutes': [,],
+        'Last 1 hour': [,],
+        'Last 3 hours': [,],
+        'Last 6 hours': [,],
+        'Last 12 hours': [,],
+        'Last 24 hours': [,],
+        'Last 2 days': [,],
+        'Last 7 days': [,],
+        'Last 30 days': [,],
+        'Last 90 days': [,],
+        'Last 6 months': [,],
+        'Last 1 year': [,],
+        'Today': [,],
+        'Yesterday': [,],
+        'This week': [,],
+        'This month': [,],
+        'Last month': [,]
+      },
+      'alwaysShowCalendars': true,
+      'startDate': moment(window.MetricQWebView.instances[0].handler.startTime),
+      'endDate': moment(window.MetricQWebView.instances[0].handler.stopTime)
+    }, function (start, end, label) {
+      if (Math.abs(moment().unix() - end.unix()) < 60 * 2) {
```

Comment:
Ah, so this is the magic. If the end date is in the last 2 minutes, it's detected as being a relative timestamp.

That's a rare occasion where I would suggest a comment in the code to explain this.

```suggestion
      // Unfortuantely we cannot distinguish relative and absolute timestamps.
      // So instead we use a heuristic: every time range that ends within the last two minues is considered relative to now
      if (Math.abs(moment().unix() - end.unix()) < 60 * 2) {
```